### DISCUSSION TITLE
How to specify the password of a user when creating a vault user using adduser command of security-proxy-setup serrvice ?

### DISCUSSION THREAD

**Discussion by younes199511** [edgexfoundry/community#152] [Q&A] [answered]
Hello,
Is it possible to specify the vault user password when creating it using adduser command of security-proxy-setup service in EdgeX 3.0 ? 

Currently when i execute the command adduser as described in the [documentation](https://docs.edgexfoundry.org/3.0/security/Ch-Authenticating/) it generates a random password for the new user and i'm wondering if it is possible to specify the user's password as an argument of the adduser command ( in addition to the jwtTTL and tokenTTL) ? I took a look into the [code](https://github.com/edgexfoundry/edgex-go/blob/main/internal/security/config/command/proxy/adduser/command.go) and it seems that the adduser has no password parameter. 

Thank you

**Comment by bnevis-i** [comment, marked as answer]
> Is it possible to specify the vault user password when creating it using adduser command of security-proxy-setup service in EdgeX 3.0 ?

No, it is not.  If you are unhappy with that, there is nothing stopping you from making a custom version of the security-proxy-setup with the desired functionality.  Note that Vault's login method is exposed through the reverse proxy, so if you do allow custom passwords, they would be subject to dictionary attacks and credential-stuffing attacks.

Also note that Vault added some brute force protection in 1.13... https://support.hashicorp.com/hc/en-us/articles/16772116543507-Prevent-Vault-from-Brute-Force-Attack-User-Lockout   This is a brand-new feature in Vault.

Note that as mentioned in the (documentation)[https://docs.edgexfoundry.org/3.0/security/Ch-Authenticating/] it is possible with more extensive customizations to the system to enable enterprise ID providers as well (left as an exercise to the adopter).
